tls/s2n_server_new_session_ticket.c

```diff
@@ -253,3 +253,76 @@ int s2n_tls13_server_nst_write(struct s2n_connection *conn, struct s2n_stuffer *
 
     return S2N_SUCCESS;
 }
+
+/**
+ *= https://tools.ietf.org/rfc/rfc8446#section-4.6.1
+ *#     struct {
+ *#         uint32 ticket_lifetime;
+ *#         uint32 ticket_age_add;
+ *#         opaque ticket_nonce<0..255>;
+ *#         opaque ticket<1..2^16-1>;
+ *#         Extension extensions<0..2^16-2>;
+ *#      } NewSessionTicket;
+**/
+int s2n_tls13_server_nst_recv(struct s2n_connection *conn, struct s2n_stuffer *input)
+{
+    POSIX_ENSURE_REF(conn);
+    POSIX_ENSURE_REF(input);
+
+    POSIX_ENSURE(conn->mode == S2N_CLIENT, S2N_ERR_SAFETY);
```

Comment:
Isn't there an S2N_ERR_CLIENT_MODE error specifically for doing things as a client that the client is not allowed to do?

Reply:
The client is allowed to do this, the server is not.

Reply:
I'll do S2N_ERR_BAD_MESSAGE as that is more appropriate